### DISCUSSION TITLE
Cannot restart Kimai after reboot

### DISCUSSION THREAD

**Discussion by malewpro** [kimai/kimai#4642] [Docker]
### Describe the problem

I rebooted the VM within which I have Kimai installed and running on Docker but have had a hard time getting Kimai to restart and am worried I have lost the data and set-up I've built. I assume this is a fairly trivial issue to resolve but I can't seem to find guidance in the documentation.

### Describe your setup and add your Docker compose file (redact your credentials)

I used the Docker Run commands listed below back when I spun up the container for the first time and Kimai worked well. When I had to reboot my VM today it didn't restart and I have not found a way to restart the container without essentially creating a fresh install. I've found the old docker volumes still in my filesystem so thankfully the data is still there. Any advice on how to recover the data and avoid this sort of issue in the future? 

Redacted Docker Run Commands:
`sudo docker run --rm --name kimai-mysql     -e MYSQL_DATABASE=kimai     -e MYSQL_USER=[redacted]     -e MYSQL_PASSWORD=[redacted]     -e MYSQL_ROOT_PASSWORD=[redacted]     -p 3399:3306 -d mysql`

`sudo docker run --rm --name kimai -ti -p 8001:8001 -e TRUSTED_HOSTS=192.168.1.25,localhost,127.0.0.1 -e ADMINMAIL=[redacted] -e ADMINPASS=[redacted] -e DATABASE_URL=mysql://[redacted]:[redacted]@192.168.1.25:3399/[redacted] kimai/kimai2:apache`

My working environment is as follows:
- Ubuntu VM running Docker version 24.0.7

**Comment by empx92** [comment]
Im not an expert with your actual problem but in my case i use portainer to monitor my containers. in that web-interface i get all informations i need. i also start or restart containers with portainer. i did set up my kimai with a docker-compose.yml so i didn't ever had a problem like you described. And our server also needs some restarts from time to time ;) so maybe you should be sure that your actual setup is working and alright to prevent such problems

**Comment by Not-Code** [comment]
Hi @malewpro,my first suggestion is to move from imperative commands to declarative commands. If you don't know what I'm talking about just start using docker-compose and you will thank me in a few months or so.

Now, on your specific issues here a brief explanation.

`--rm`
This [option](https://docs.docker.com/engine/reference/commandline/container_run/#rm) should delete containers and anonymous related volumes on exits. And when a machine restarts the container should go to an exited state, deleting everything (or at least this is what happens to me locally with podman).

If instead you are 100% that the volume you are listing in docker was referring to kimai-mysql, tell me and we can try to retrieve the content.

@tobybatch 
I think that we are partly to blame here. 
There's no way that an unexpert user can do the right configuration if he goes to the [docker](https://www.kimai.org/documentation/docker.html) section of the documentation. I think that we should add a disclaimer in the docker section explaining that the below commands are just for a quick preview of Kimai and they should not be used for any kind of deployment.
Moving the user attention to the [compose](https://www.kimai.org/documentation/docker-compose.html) section instead.
What do you think?

**Reply by kevinpapst** [reply]
The docs clearly say:

> Keep in mind that this Docker setup is transient and the data will disappear when you remove the containers.

I can add some colors to emphasize the warning. 
And I will update the docs if someone tells me what & how (remember: I don't use docker).

**Reply by Not-Code** [reply]
True, and I'm the second person in this conversation that probably hasn't read that 🤣 .
Should we make it more visible like we did for the nginx-fpm-reverse-proxy?

<img width="1359" alt="image" src="https://github.com/kimai/kimai/assets/18069705/f733ad34-a7cf-484f-b351-8be851ae0ee0">

**Reply by Not-Code** [reply]
> I can add some colors to emphasize the warning.

Yeah, I think that emphasizing the warning with colors should be enough to get attention.

**Reply by kevinpapst** [reply]
Done. Changed wording, added warning color, moved warning to start of chapter, add warning for the `docker rm` command. Thanks for bringing that up @Not-Code 👍 

And added a link at the end that says:

> If you are happy with Kimai, you can now setup your Docker installation using Docker Compose

I hope that funnel is understandable. We can't help if people don't want to read the docs 😁

**Reply by malewpro** [reply]
Hey all, thanks for getting back to me on this. I did migrate to a Docker Compose implementation now, which I generally prefer. I'm not sure how I missed that as an initial set up option. I've given up on trying to salvage the data from my initial instance. I appreciate the change to the documentation. Kimai has been great, this is just a hiccup and hopefully something others can avoid in the future. Really appreciate the response here!

**Reply by malewpro** [reply]
Happy to close this discussion btw. I'll leave that to you all, though. Not sure if there's more to discuss here re: documentation.

**Reply by kevinpapst** [reply]
Unfortunately such documentation shortages often only become visible after an incident.

Sorry for your initial experience @malewpro and thanks for your kind feedback.
